scripts/generate_xlsx.py

```diff
@@ -0,0 +1,374 @@
+        self.slot_def = slot_def
+        self.schema = schema
+        self.prefix_map = prefix_map
+        self.range = slot_def.range if slot_def.range else "string"
+        self.cls_name = str(range) if range in schema.all_classes() else None
+        self.enum_name = str(range) if range in schema.all_enums() else None
+        self.type_name = str(range) if range in schema.all_types() else None
+
+
+def _make_value_cell(ws, fill_content):
+    """Creates a new cell for the value area of the worksheet"""
+    value_cell = Cell(ws)
+    if fill_content:
+        value_cell.fill = fill_content
+    value_cell.border = THIN_BORDER_GRAY
+    return value_cell
+
+
+def _build_prefix_map(wb_config: WorkbookConfig) -> dict[str, str]:
+    """Builds a dictionary mapping every class to the configured column header prefix"""
+    return {
+        ws_config.class_name: ws_config.prefix for ws_config in wb_config.worksheets
+    }
+
+
+def _get_ordered_slots(
+    schema: SchemaView,
+    slot_order: list[str],
+    cls_name: str,
+    prefix_map: dict[str, str],
+) -> list[SlotDefinition]:
+    """Given a class, generates a list of slots that must be rendered. Slots
+    that are listed in the slot_order config parameter are at the top of the
+    list in their respective order. Slots that are referencing other classes
+    which are not part of this workbook are omitted."""
+    slots = schema.class_induced_slots(cls_name)
+    slot_names = {slot.name: idx for idx, slot in enumerate(slots)}
+    ordered = [
+        slots[slot_names[slot_name]]
+        for slot_name in slot_order
+        if slot_name in slot_names
+    ]
+    unordered = [slot for slot in slots if slot.name not in slot_order]
+    all_slots = ordered + unordered
+    # We ignore slots which have a class range when the target class is not included in this workbook.
+    ignored_slots = [
+        slot
+        for slot in all_slots
+        if slot.range in schema.all_classes()
+        and slot.range not in prefix_map
+        and schema.get_identifier_slot(slot.range)
+    ]
+    # If one of those is mandatory, we raise an error
+    for ignored_slot in ignored_slots:
+        if ignored_slot.required:
+            raise RuntimeError(
+                f"Slot '{cls_name}.{ignored_slot.name}' is mandatory, but target class is not included in the workbook!"
+            )
+    slots = [slot for slot in all_slots if slot not in ignored_slots]
+    return slots
+
+
+def create_xlsx_files(config_path: Path, out_dir: Path):
+    """Creates the XLSX workbooks as configured in the provided configuration
+    and writes them to the specified output path"""
+    with open(config_path, "r", encoding="utf8") as config_file:
+        config = Config.parse_obj(yaml.safe_load(config_file))
+
+    # Read schema
+    schema = SchemaView(str(SCHEMA_PATH))
+
+    # Create the workbooks
+    for wb_config in config.workbooks:
+        print(wb_config.file_name, end="", flush=True)
```

Comment:
Did you leave the print statement here on purpose?

Reply:
Yes that's intentional, it prints the progress